qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,312 @@
+            TypeError: If the provided minimum eigensolver does not support auxiliary operators.
+        """
+        if max_vars_per_qubit not in (1, 2, 3):
+            raise ValueError("max_vars_per_qubit must be 1, 2, or 3, but was {max_vars_per_qubit}.")
+        self._max_vars_per_qubit = max_vars_per_qubit
+        self.min_eigen_solver = min_eigen_solver
+        # Use ``QuadraticProgramToQubo`` to convert the problem to a QUBO.
+        if rounding_scheme is None:
+            rounding_scheme = SemideterministicRounding()
+        self._rounding_scheme = rounding_scheme
+        self._converters = QuadraticProgramToQubo(
+            penalty=penalty,
+        )
+
+    @property
+    def min_eigen_solver(self) -> MinimumEigensolver:
+        """Return the minimum eigensolver."""
+        return self._min_eigen_solver
+
+    @min_eigen_solver.setter
+    def min_eigen_solver(self, min_eigen_solver: MinimumEigensolver) -> None:
+        """Set the minimum eigensolver."""
+        if not min_eigen_solver.supports_aux_operators():
+            raise TypeError(
+                f"The provided MinimumEigensolver ({type(min_eigen_solver)}) "
+                "does not support auxiliary operators."
+            )
+        self._min_eigen_solver = min_eigen_solver
+
+    @property
+    def max_vars_per_qubit(self) -> int:
+        """Return the maximum number of variables per qubit."""
+        return self._max_vars_per_qubit
+
+    @max_vars_per_qubit.setter
+    def max_vars_per_qubit(self, max_vars_per_qubit: int) -> None:
+        """Set the maximum number of variables per qubit."""
+        self._max_vars_per_qubit = max_vars_per_qubit
+
+    @property
+    def rounding_scheme(self) -> RoundingScheme:
+        """Return the rounding scheme."""
+        return self._rounding_scheme
+
+    @rounding_scheme.setter
+    def rounding_scheme(self, rounding_scheme: RoundingScheme) -> None:
+        """Set the rounding scheme."""
+        self._rounding_scheme = rounding_scheme
+
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with this optimizer.
+
+        Checks whether the given problem is compatible, i.e., whether the problem can be converted
+        to a QUBO, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            A message describing the incompatibility.
+        """
+        return QuadraticProgramToQubo.get_compatibility_msg(problem)
+
+    def solve_relaxed(
+        self,
+        encoding: QuantumRandomAccessEncoding,
+    ) -> tuple[MinimumEigensolverResult, RoundingContext]:
+        """Solve the relaxed Hamiltonian given by the encoding.
+
+        Args:
+            encoding: The ``QuantumRandomAccessEncoding``, which must have already been ``encode()``ed
+                with a ``QuadraticProgram``.
```

Comment:
This does not come out well. I think the double ticks in the middle of encode()ed does not get recognized
![image](https://github.com/qiskit-community/qiskit-optimization/assets/40241007/a98713d3-151e-4159-93a9-a00aa0c7ad9c)
Maybe these can go over to class and method links anyway, at least the encode() and QuadraticProgram. The QuantumRandomAccessEncoding it does not need it as much since its right there in the parameter type. Maybe it could be said instead more like
 ``encoding: An encoding instance for which :meth:`~QuantumRandomAccessEncoding.encode` has already been called so it has been encoded with a :class:`QuadraticProgram` `` - probably the refs need fixing from what I show but something like that perhaps

Reply:
updated with your suggestion.